### PR TITLE
Issue1198: error reading deleted secret

### PR DESCRIPTION
**Pull request by eikenb**

### CODE REVIEW

File: dependency/dependency_test.go [hunk 1]
```diff
@@ -149,6 +149,16 @@ func (v *vaultServer) CreateSecret(path string, data map[string]interface{},
 	return err
 }
 
+// deleteSecret lets us delete keys as needed for tests
+func (v *vaultServer) deleteSecret(path string) error {
+	path = v.secretsPath + "/" + path
```
Comment (adams-sarah) [inline dependency/dependency_test.go]: maybe better not to modify the input? maybe instantiate a new variable? or else maybe all on one line like:
```
/* blah */.Delete(v.secretsPath + "/" + path)
```

File: dependency/vault_read_test.go [hunk 2]
```diff
@@ -325,6 +326,34 @@ func TestVaultReadQuery_Fetch_KVv2(t *testing.T) {
...
+		d, err := NewVaultReadQuery(path)
+		if err != nil {
+			t.Fatal(err)
+		}
+		_, _, err = d.Fetch(clients, nil)
+		if err == nil {
+			t.Fatal("Nil received when error expected")
+		}
+		exp_err := fmt.Errorf("no secret exists at %s", path)
+		assert.Equal(t, exp_err, errors.Cause(err))
```
Comment (adams-sarah) [inline dependency/vault_read_test.go]: regardless of whether the team prefers using `assert` or `t.Fatal` for tests, we should at least keep it consistent within a test. I'd like to see here instead, `if exp_err != errors.Cause(err); t.Fatal` or else a change in the rest of the test assertions. wdyt?

Comment (eikenb) [reply dependency/vault_read_test.go]: Yeah... I was going back and forth about using assert here and ended up in between. Assert is used throughout the tests and so I thought staying consistent would be good, but being more idiomatic would also be good.

I'm going to try the more idiomatic approach and see how it works going forward.

File: dependency/dependency_test.go (hunk 1 above)
Comment (eikenb) [reply dependency/dependency_test.go]: No variable at all is better IMO.  Thanks!

File: dependency/vault_read_test.go (hunk 2 above)
Comment (eikenb) [reply dependency/vault_read_test.go]: Ended up with a string comparison for the error compare. Hope the 1.13 errors changes help with this.

**Pull request merged**